### PR TITLE
edit: 20160425 -> 20180228 and fix darwin build

### PR DESCRIPTION
**Pull request by siraben**
###### Motivation for this change
Also fixed darwin build on one of its dependencies, cwebbin.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 114857` at 6b3869d0 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/c0f2ddcff083e0879ef69e027f2d89b0">cwebbin</a></li></li>
    <li><a href="https://gist.github.com/02c480fd7398bdaee561d1d08dc69d88">edit</a></li></li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/development/tools/misc/cwebbin/default.nix:41:3:
```
   |
41 |   buildPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/editors/edit/default.nix:20:3:
```
   |
20 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/tools/misc/cwebbin/default.nix:47:3:
```
   |
47 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

**Comment by siraben** [comment]
Builds and runs on macOS, able to edit the buffer.

**Comment by SuperSandro2000** [comment]
``nix-build -E "with import ./. {}; pkgsCross.aarch64-multiplatform.edit"``

https://termbin.com/imh8

**Comment by siraben** [comment]
> ``nix-build -E "with import ./. {}; pkgsCross.aarch64-multiplatform.edit"``
> 
> 
> 
> https://termbin.com/imh8

Yeah wasn't able to test cross-compilation.

File: pkgs/applications/editors/edit/default.nix [hunk 1]
```diff
@@ -2,39 +2,31 @@
...
 
   installPhase = ''
-    mkdir -p $out/bin/
-    cp obj/edit $out/bin/edit
+    install -Dm755 obj/edit -t $out/bin
   '';
 
   meta = with lib; {
     description = "A relaxing mix of Vi and ACME";
     homepage = "http://c9x.me/edit";
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/edit/default.nix:26]:
```suggestion
    homepage = "https://c9x.me/edit";
```

**Pull request merged**